diff -Nur schroot-1.0.4.orig/configure.ac schroot-1.0.4.chng/configure.ac
--- schroot-1.0.4.orig/configure.ac	2006-10-21 12:26:59.000000000 +0200
+++ schroot-1.0.4.chng/configure.ac	2007-01-27 21:34:34.000000000 +0100
@@ -148,34 +148,7 @@
 	        AC_MSG_FAILURE([libboost_program_options (Boost C++ Libraries) is not installed, but is required by schroot])])
 LDFLAGS="${saved_ldflags}"
 
-AC_MSG_CHECKING([for boost::program_options::options_description::options() in -lboost_program_options])
-saved_ldflags="${LDFLAGS}"
-LDFLAGS="${LDFLAGS} -lboost_program_options"
-AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <boost/program_options.hpp>],
-                               [boost::program_options::options_description testgrp("test group");
-                                bool notused = testgrp.options().empty();
-])],
-               [AC_MSG_RESULT([yes])
-                BOOST_PROGRAM_OPTIONS_DESCRIPTION_METHODS="current"],
-               [AC_MSG_RESULT([no])
-                BOOST_PROGRAM_OPTIONS_DESCRIPTION_METHODS="old"])
-LDFLAGS="${saved_ldflags}"
-AH_TEMPLATE(BOOST_PROGRAM_OPTIONS_DESCRIPTION_OLD, [Set if boost::program_options::options_description::options() is not available])
-if test "$BOOST_PROGRAM_OPTIONS_DESCRIPTION_METHODS" = "old"; then
-  AC_DEFINE(BOOST_PROGRAM_OPTIONS_DESCRIPTION_OLD, 1)
-fi
-
-AC_MSG_CHECKING([for boost::regex in -lboost_regex])
-saved_ldflags="${LDFLAGS}"
-LDFLAGS="${LDFLAGS} -lboost_regex"
-AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <boost/regex.hpp>],
-                               [boost::regex("^foo[bar]$")])],
-               [AC_MSG_RESULT([yes])
-	        BOOST_LIBS="${BOOST_LIBS} -lboost_regex"],
-               [AC_MSG_RESULT([no])
-	        AC_MSG_FAILURE([libboost_regex (Boost C++ Libraries) is not installed, but is required by schroot])])
-LDFLAGS="${saved_ldflags}"
-
+BOOST_LIBS="${BOOST_LIBS} -lboost_regex"
 AC_SUBST([BOOST_LIBS])
 
 AC_MSG_CHECKING([__gnu_cxx::stdio_filebuf construction semantics])
